File: Tree.js
class Tree{
    constructor(root){
        this.leftLeaf = null
        this.root = root
        this.rightLeaf = null
    }
    insert(data){
        if(data > this.root){
            if(this.rightLeaf == null){
                this.rightLeaf = new Tree(data)
            }
            if(this.rightLeaf != null){
                this.rightLeaf.insert(data)
            }
        }
        if(data < this.root){
            if(this.leftLeaf == null){
                this.leftLeaf = new Tree(data)
            }
            if(this.leftLeaf != null){
                this.leftLeaf.insert(data)
            }
        }
    }
    print(counter){
        counter += 4
        var tab = ' '
        if(this.leftLeaf !== null){
            this.leftLeaf.print(counter)
        }
        console.log(tab.repeat(counter) + `${this.root}`)
        if(this.rightLeaf !== null){
            this.rightLeaf.print(counter)
        }
    }

    find(number){
        if(number > this.root){
            if(this.rightLeaf !== null){
                this.rightLeaf.find(number)
            }
            else{
                console.log(`${number} does not exist`)
            }
        }
        if(number === this.root){
            console.log(`${number} exist`)
        }
        if(number < this.root){
            if(this.leftLeaf !== null){
                this.leftLeaf.find(number)
            }
            else{
                console.log(`${number} does not exist`)
            }
        }
    }
}

function getRandomInt(min, max) {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min)) + min;
}

function generateArray(min, max){
    count = max - min
    array = []
    while(count){
        number = getRandomInt(min, max)
        if(array.includes(number)){}
        else{
            array.push(number)
            count--
        }
    }
    return array
}


array = generateArray(1,15)
const tree = new Tree(array[array.length - 1])
array.pop()
for(var i = 0; i < array.length; i++){
    tree.insert(array[i])
}
tree.print(0)
tree.find(20)

File: Tree.py
import random
class Tree:
    def __init__(self, node:int = 0):
        self.root = node
        self.left = None
        self.right = None
    
    def insertData(self, number:int):
        if number > self.root:
            if self.right == None:
                self.right = Tree(number)
            if self.right != None:
                self.right.insertData(number=number)
        if number < self.root:
            if self.left == None:
                self.left = Tree(number)
            if self.left != None:
                self.left.insertData(number=number)
    
    def findData(self, number:int = 0):
        if number > self.root:
            self.right.findData(number)
        if number < self.root:
            self.left.findData(number)
        if number == self.root:
            print("Element was found")

    def printTree(self, counter:int = 0):
        counter += 4
        tab = ' '*counter
        if self.left:
            self.left.printTree(counter=counter)
        print(tab, self.root)
        if self.right:
            self.right.printTree(counter=counter)

array = [random.randint(1,15) for i in range(10)]
print(array)
tree = Tree(array[0])
array.pop(0)
for elem in array:
    tree.insertData(elem)
tree.printTree()
tree.findData(7)
